x-pack/plugins/fleet/server/services/agent_policy.ts

```diff
@@ -1070,6 +1080,104 @@ class AgentPolicyService {
     };
   }
 
+  public async createAgentlessAgent(
+    esClient: ElasticsearchClient,
+    soClient: SavedObjectsClientContract,
+    agentlessAgentPolicy: AgentPolicy
+  ) {
+    if (!appContextService.getCloud()?.isCloudEnabled) {
+      throw new AgentlessAgentCreateError('Agentless agent not supported');
+    }
+    if (!agentlessAgentPolicy.supports_agentless) {
+      throw new AgentlessAgentCreateError('Agentless agent policy does not have agentless enabled');
+    }
+
+    const agentlessConfig = appContextService.getConfig()?.agentless;
+    if (!agentlessConfig) {
+      throw new AgentlessAgentCreateError('missing agentless configuration');
+    }
+
+    const policyId = agentlessAgentPolicy.id;
+    const { fleetUrl, fleetToken } = await this.getFleetUrlAndTokenForAgentlessAgent(
+      esClient,
+      policyId,
+      soClient
+    );
+
+    const tlsConfig = new SslConfig(
+      sslSchema.validate({
+        enabled: true,
+        certificate: agentlessConfig.api.tls.certificate,
+        key: agentlessConfig.api.tls.key,
+        certificateAuthorities: agentlessConfig.api.tls.ca,
+      })
+    );
+
+    try {
+      const { data, status, statusText } = await axios<AgentlessApiResponse>({
+        url: `${agentlessConfig.api.url}/deployments`,
+        data: {
+          policy_id: policyId,
+          fleet_url: fleetUrl,
+          fleet_token: fleetToken,
+          stack_version: appContextService.getKibanaVersion(),
+        },
+        method: 'POST',
+        headers: {
+          'Content-type': 'application/json',
+        },
+        httpsAgent: new https.Agent({
+          rejectUnauthorized: tlsConfig.rejectUnauthorized,
+          cert: tlsConfig.certificate,
+          key: tlsConfig.key,
+          ca: tlsConfig.certificateAuthorities,
+        }),
+      });
+
+      // if status is not CREATED, throw an error
+      if (status !== 200) {
```

Comment:
```suggestion
      if (status !== 200 || status !== 201) {
```

Seems that we don't report the correct status. Since both 200 and 201 are success codes, I'd be more permissive here.

Is there no way to handle success vs failure without specifying specific http codes? I found https://www.geeksforgeeks.org/which-http-response-status-codes-result-in-then-and-which-in-catch/.

Reply:
@orestisfl,  this is funny because I originally had it checking for 201 status.  The new change should handle all statuses that are not in the 2xx as errors otherwise it returns the response data.

Reply:
> The new change should handle all statuses not in the 2xx as errors otherwise it returns the response data.

I don't see any change in the status code handling :thinking:

Reply:
Ack!  Apologies,  my blind push failed because I rebased. Should be good now.